src/Compilers/VisualBasic/Portable/Symbols/TypeSymbolExtensions.vb

```diff
@@ -1308,6 +1308,17 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
                 namedTypeSymbol.IsContainedInNamespace(NameOf(System), NameOf(System.Threading))
         End Function
 
+        ' Keep in sync with C# equivalent.
+        <Extension>
+        Friend Function IsWellKnownTypeUnmanagedType(typeSymbol As TypeSymbol) As Boolean
+            Dim namedTypeSymbol = TryCast(typeSymbol, NamedTypeSymbol)
+            Return namedTypeSymbol IsNot Nothing AndAlso
+                namedTypeSymbol.Name = "UnmanagedType" AndAlso
+                namedTypeSymbol.Arity = 0 AndAlso
```

Comment:
>namedTypeSymbol.Arity = 0 AndAlso [](http://example.com/codeflow?start=16&length=33)

It looks like the C# version and related C# extension methods are not checking `Arity == 0`. Should we add that check to those methods?

Reply:
Indeed, added a test and opened an issue to follow up - #75681